api-compose-object.go

```diff
@@ -210,16 +211,38 @@ func (c Client) copyObjectDo(ctx context.Context, srcBucket, srcObject, destBuck
 	for k, v := range metadata {
 		headers.Set(k, v)
 	}
+	if !dstOpts.Internal.ReplicationStatus.Empty() {
+		headers.Set(amzBucketReplicationStatus, string(dstOpts.Internal.ReplicationStatus))
+	}
+	if !dstOpts.Internal.SourceMTime.IsZero() {
+		headers.Set(minIOBucketSourceMTime, dstOpts.Internal.SourceMTime.Format(time.RFC3339))
+	}
+	if dstOpts.Internal.SourceETag != "" {
+		headers.Set(minIOBucketSourceETag, dstOpts.Internal.SourceETag)
+	}
+	if len(dstOpts.UserTags) != 0 {
+		headers.Set(amzTaggingHeader, s3utils.TagEncode(dstOpts.UserTags))
+	}
+
+	reqMetadata := requestMetadata{
+		bucketName:   destBucket,
+		objectName:   destObject,
+		customHeader: headers,
+	}
+	if dstOpts.Internal.SourceVersionID != "" {
+		if _, err := uuid.Parse(dstOpts.Internal.SourceVersionID); err != nil {
+			return ObjectInfo{}, errInvalidArgument(err.Error())
+		}
+		urlValues := make(url.Values)
+		urlValues.Set("versionId", dstOpts.Internal.SourceVersionID)
```

Comment:
LGTM,

Otherwise, is this style already released ? Why not using 'minIOBucketSourceVersionID' for consistency with others, also this is more prone to fail with AWS, probably in other use cases in the future.

Reply:
all other API's use a query string for version id - so trying to stay consistent with that. With AWS, this version id would never be set